shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -101,195 +101,58 @@ def _generate_state_file_string(self, gcp_backend_resource: GcpTfBackend):
-                        # if json_path:
-                        # logger.exception(f"json_path: {clp_json_path}")
-                        # json = "C:\\Users\\CSadmin123456\\CS\\tf-gcp\\prod.json"
-                        # logger.exception(f"def: {json}")
-                        os.environ[GOOGLE_APPLICATION_CREDENTIALS] = clp_json_path
-                        # self._handle_exception_logging(logger, f"CLP: {clp_json_path}  DEF: {json}") 
-                        # os.environ[GOOGLE_APPLICATION_CREDENTIALS] = "C:\\Users\\CSadmin123456\\CS\\tf-gcp\\prod.json"
-            except Exception as e:
-                self._handle_exception_logging(logger, f"There was an issue accessing GCP. {e}")
-            bucket_name = gcp_backend_resource.bucket_name
-            if bucket_name:      
-                self._validate_bucket_exists(bucket_name, context)
-
-    def _get_attrbiute_value_from_clp(self, attributes, model_prefix, attribute_name) -> str:
-        for attribute in attributes:
-
-            if attribute.Name == f"{model_prefix}{attribute_name}":
-                return attribute.Value
-        return ""
-
+    def _create_gcp_session(self, context, project_id, logger):
+        # with LoggingSessionContext(context) as logger:
+        api = CloudShellSessionContext(context).get_api()
+        gcp_backend_resource = GcpTfBackend.create_from_context(context)
+        # try:
+        private_key = api.DecryptPassword(gcp_backend_resource.private_key).Value.replace("\\n", "\n")
+        email = api.DecryptPassword(gcp_backend_resource.client_email).Value
+        if not project_id:
+            self._raise_and_log(logger, "Project id must be filled")
+        # Key and email defines on GCP TF BACKEND RESOURCE
+        if private_key and email:
+            if gcp_backend_resource.cloud_provider:
+                self._raise_and_log(logger, "Only one method of authentication should be filled")
+            with open(DYNAMIC_JSON, 'w', encoding='utf-8') as f:
+                json.dump({ "type": TYPE_ACCOUNT, "project_id": project_id, "private_key": private_key, "client_email": email,"auth_uri": AUTH_URI, "token_uri": TOKEN_URI }, f, ensure_ascii=False, indent=4)
+            os.environ[GOOGLE_APPLICATION_CREDENTIALS] = DYNAMIC_JSON
+        # Keys not defines on GCP TF BACKEND RESOURCE (CLP reference should have been set)
+        else:
+            # CLP had not been set...
+            if not gcp_backend_resource.cloud_provider:
+                self._raise_and_log(logger, "At least one method of authentication should be filled")
+            # Check a correct CLP has been reference
+            clp_details = api.GetResourceDetails
+            clp_resource_details = self._get_and_validate_clp(clp_details, gcp_backend_resource, logger)
+            if clp_resource_details.ResourceModelName == GCP1G_MODEL:
+                myactual=clp_details(clp_resource_details.Name)
```

Comment:
this looks redundant to me because the ResourceInfo object is already returned from "_get_and_validate_clp"